### PR TITLE
[TIMOB-25678] Android: Amend hasActivityTransitions() for transitions

### PR DESCRIPTION
**Pull request by garymathews**
- Fix `hasActivityTransitions()` to allow transitions when a `sharedElement` has not been specified

###### TEST CASE
```JS
var win = Ti.UI.createWindow({
        backgroundColor: 'blue',
        layout: 'vertical'
    }),
    lbl = Ti.UI.createLabel({
        left: 5,
        text: 'TEST LABEL',
        transitionName: 'lbl'
    }),
    btn = Ti.UI.createButton({
        title: 'OPEN'
    });

btn.addEventListener('click', function () {
    var win2 = Ti.UI.createWindow({
            backgroundColor: 'red',
            layout: 'vertical',
            activityEnterTransition: Titanium.UI.Android.TRANSITION_SLIDE_RIGHT
        }),
        lbl2 = Ti.UI.createLabel({
            right: 5,
            text: 'TEST LABEL',
            transitionName: 'lbl'
        }),
        btn2 = Ti.UI.createButton({
            title: 'CLOSE'
        });

    btn2.addEventListener('click', function () {
        win2.close();
    });

    win2.add([lbl2, btn2]);
    // win2.addSharedElement(lbl, 'lbl'); // this should not need uncommenting to see a Window transition

    win2.open();
});

win.add([lbl, btn]);
win.open();
```

[JIRA Ticket](https://jira.appcelerator.org/browse/TIMOB-25678)

### CODE REVIEW

**Comment by jquick-axway** [comment]
I'm still running into that bug where the window is transparent when it shouldn't be.
I tested it on my Android 8.1 Pixel XL device.

@lokeshchdhry, you want to give it a go too?

**Comment by jquick-axway** [comment]
I've tested this with several of my own test apps on real Android 9.0, 8.1, and 4.4 devices. It works fine.

I did notice that the 300ms delay happens every time on app startup since property "animated" defaults to `true`. Subsequent windows open without the delay of course. But the 1st window it happens every time. Although it's not that noticeable in my opinion.

**Comment by sgtcoolguy** [comment]
@jquick-axway Is this approved then? Can we give it an approving review and tag it in-qe-testing? (the back port is scheduled for 8.0.1)

**Comment by jquick-axway** [comment]
This PR still has a 300 ms delay which it adds by default. So, let's not.

File: android/modules/ui/src/java/ti/modules/titanium/ui/WindowProxy.java [hunk 1]
```diff
@@ -137,21 +137,37 @@ public void close(@Kroll.argument(optional = true) Object arg)
...
+		if (animated && topActivity instanceof TiBaseActivity) {
+			final TiBaseActivity baseActivity = (TiBaseActivity) topActivity;
+			if (!baseActivity.isResumed) {
+				getMainHandler().postDelayed(new Runnable() {
+					@Override
+					public void run()
+					{
+						handleOpen(options);
+					}
+				}, 300);
```
Comment (garymathews) [inline android/modules/ui/src/java/ti/modules/titanium/ui/WindowProxy.java]: This may only be necessary on lower Android OS versions, need to revisit.

**Review by jquick-axway — APPROVED** [review]
CR: Pass

**Comment by lokeshchdhry** [comment]
FR Passed.

Transitions works as expected when sharedElement has not been specified.

FYI : Transitions are not supported on Android 4.1

Studio Ver:     5.1.2.201903111843
SDK Ver:        8.1.0 local build
OS Ver:         10.14
Xcode Ver:      Xcode 10.1
Appc NPM:       4.2.13
Appc CLI:       7.0.10
Daemon Ver:     1.1.3
Ti CLI Ver:     5.1.1
Alloy Ver:      1.13.9
Node Ver:       8.9.1
NPM Ver:        5.5.1
Java Ver:       10.0.2
Devices:        ⇨ google Nexus 5 (Android 6.0.1)
                ⇨ google Pixel (Android 9)

**Pull request merged**